parse_result.py

```diff
@@ -130,13 +154,16 @@ def static():
             process(job)
 
     static_tests = result_dict.pop("static_tests", {})
-    if nfailed:
-        print("--- result: BUILD FAILED!")
-        if html and ((nfailed > 1) or has_passed(static_tests)):
-            print("\n--- ", end="")
-            print(html_link("#error0", "JUMP TO FIRST ERROR OUTPUT"))
-    else:
-        print("--- result: BUILD SUCCESSFUL.")
+    with open("badge.svg", "w") as badge_svg:
```

Comment:
How about factoring out the badge writing?

```
def badge_write(filename, status="unknown"):
 with open(filename, "w") as badge_svg:
   badge_svg.write(badge.substitute(status=status))
```

... then call that within ```nfailed``` or else?
Feels weird to have the stdout printing within the "with".

Reply:
Done.